R/mlp-fit.R

```diff
@@ -636,14 +659,16 @@ mlp_fit_imp <-
 
       # training loop
       coro::loop(
-        for (batch in dl) {
-          pred <- model(batch$x)
-          loss <- loss_fn(pred, batch$y, class_weights)
-
-          optimizer$zero_grad()
-          loss$backward()
-          optimizer$step()
+       for (batch in dl) {
```

Comment:
The new code is the same as the internals to linear and logistic regression. @dfalbel Is this a good change to make for neural networks too?

Reply:
Yes, this is a good change. This is actually required if using second order optimization algorithms like LBFGS.